tests/unit/test.program.js

```diff
@@ -841,6 +841,42 @@ describe('program.main', () => {
     }
   );
 
+  describe('--no-input', () => {
+    const fakeCommands = fake(commands, {
+      run: () => Promise.resolve(),
+    });
+
+    const testCases = [
+      ['--no-input', {noInput: true}],
+      ['--no-input=false', {noInput: false}],
+      ['--no-input=true', {noInput: true}],
+      ['--input', {noInput: false}],
+      ['--input=false', {noInput: true}],
+      ['--input=true', {noInput: false}],
+    ];
+
+    for (const [cliArg, expectedParsed] of testCases) {
+      it(`does parse ${cliArg} into the expected command options`, async () => {
+        await execProgram(['run', cliArg], {commands: fakeCommands});
+        sinon.assert.calledWithMatch(
+          fakeCommands.run,
+          expectedParsed
+        );
+        fakeCommands.run.resetHistory();
+      });
+    }
+
+    it('does set noInput to undefined if --no-input is not specified',
+       async () => {
+         fakeCommands.run.resetHistory();
```

Comment:
Nit: For consistency with the whole test in this patch, either put `resetHistory` before `execProgram`, or after it. But don't mix them.

Why did you have to create a new `it` though? Can't you use `['-v', {noInput: undefined}],` ?

Reply:
> Why did you have to create a new `it` though? Can't you use `['-v', {noInput: undefined}],` ?
yeah, that's nicer :+1: